File: 백준/Gold/3020. 개똥벌레/개똥벌레.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

/**
 * imos 써보기
 */
public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringTokenizer st;
	
	public static void main(String[] args) throws IOException {
		st = new StringTokenizer(br.readLine());
		int N = Integer.parseInt(st.nextToken());
		int H = Integer.parseInt(st.nextToken());
		
		// 양 옆에 패딩을 주기
		int[] imo = new int[H+2];
		
		// 홀수 짝수 따라서 imos 다르게 넣기
		for (int i = 0; i < N; i++) {
			int h = Integer.parseInt(br.readLine());
			if (i % 2 == 0) {
				imo[1]++;
				imo[h+1]--;
			} else {
				imo[H-h+1]++;
				imo[H+1]--;
			}
		}
		
		// 누적합 해주기
		for (int i = 1; i < H+2; i++) {
			imo[i] += imo[i-1];
		}
		
		int min = 200_000;
		int cnt = 0;
		// 1부터 H까지 min값과 cnt 구하기
		for (int i = 1; i < H+1; i++) {
			if (min > imo[i]) {
				min = imo[i];
				cnt = 1;
			} else if (min == imo[i]) {
				cnt++;
			}
		}
		
		System.out.println(min + " " + cnt);
		
	} // end of main
} // end of class

File: 백준/Gold/3020. 개똥벌레/개똥벌레.py
"""
imos 써보기
"""

import sys

input = sys.stdin.readline

N, H = map(int, input().split())

# 양 옆에 패딩 주기
imo = [0] * (H + 2)

for i in range(N):
    h = int(input())
    # 짝수인지 홀수인지 따라서 imos 다르게 표시
    if i % 2 == 0:
        imo[1] += 1
        imo[h+1] += -1
    else:
        imo[H-h+1] += 1
        imo[H+1] += -1

# 행으로 누적합 해주기
for i in range(1, H+2):
    imo[i] += imo[i-1]

# 패딩 빼기 (패딩은 0이므로)
imo = imo[1:H+1]
# 원하는 값 구하기
minValue = min(imo)
cnt = 0
for i in imo:
    if minValue == i:
       cnt += 1

print(f'{minValue} {cnt}')